### ISSUE TITLE
consistent-function-scoping: false positive with name collision

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#854]
test.js:
```js
const a = () => {
  const fn = () => "a";
  return fn();
};

const b = () => {
  const fn = () => "b";
  return fn();
};
````
.eslintrc.yaml:
````yaml
plugins:
  - unicorn

rules:
  unicorn/consistent-function-scoping: [2]
````
Output:
````
  2:17  error  Move arrow function 'fn' to the outer scope  unicorn/consistent-function-scoping
  7:17  error  Move arrow function 'fn' to the outer scope  unicorn/consistent-function-scoping
````

It's not possible to move the two `fn` to the outer scope because their names would collide.

**Comment by futpib** [comment]
Should we expect it to rename functions in addition to moving up a scope?

**Comment by silverwind** [comment]
I'd say no, there might be code that relies on `Function.name` which can make renaming impossible.

**Comment by silverwind** [comment]
Thinking about this some more, I think the proper solution is just to do the rename, even if inconvenient. `Function.name` is a obscure JS feature that should not be used or encouraged.
